model/driver/src/icon4py/model/driver/icon4py_driver.py

```diff
@@ -198,36 +183,38 @@ def _integrate_one_time_step(
         self,
         diffusion_diagnostic_state: diffusion_states.DiffusionDiagnosticState,
         solve_nonhydro_diagnostic_state: solve_nh_states.DiagnosticStateNonHydro,
-        prognostic_state_list: list[prognostics.PrognosticState],
+        prognostic_state_swp: imc_utils.Swapping[prognostics.PrognosticState],
         prep_adv: solve_nh_states.PrepAdvection,
-        inital_divdamp_fac_o2: float,
+        initial_divdamp_fac_o2: float,
         do_prep_adv: bool,
     ):
         # TODO (Chia Rui): Add update_spinup_damping here to compute divdamp_fac_o2
 
         self._do_dyn_substepping(
             solve_nonhydro_diagnostic_state,
-            prognostic_state_list,
+            prognostic_state_swp,
             prep_adv,
-            inital_divdamp_fac_o2,
+            initial_divdamp_fac_o2,
             do_prep_adv,
         )
 
         if self.diffusion.config.apply_to_horizontal_wind:
             self.diffusion.run(
-                diffusion_diagnostic_state, prognostic_state_list[self._next], self.dtime_in_seconds
+                diffusion_diagnostic_state,
+                prognostic_state_swp.other,
+                self.dtime_in_seconds,
             )
 
-        self._swap()
+        prognostic_state_swp.swap()
```

Comment:
Not 100% sure, but maybe better to introduce `Swappable` for the prognostic_state container, which has a `swap()` that calls `swap()` to all elements which have to be `Swapping`.